_posts/2023-01-04-trino-2022-the-rabbit-reflects.md

```diff
@@ -0,0 +1,581 @@
+ Javascript or Python. Martin and Dain go into great detail about how this works
+ when [answering this question at Trino Summit
+ ](https://youtu.be/mUq_h3oArp4?t=1596).
+* [ANSI SQL 2016](https://en.wikipedia.org/wiki/SQL:2016) support for JSON 
+ functions resolves a large number of issues reported by the community over the
+ years. This includes 
+ [json_array]({{site.url}}/docs/current/functions/json.html#json-array),
+ [json_object]({{site.url}}/docs/current/functions/json.html#json-object),
+ [json_exists]({{site.url}}/docs/current/functions/json.html#json-exists),
+ [json_query]({{site.url}}/docs/current/functions/json.html#json-query), and
+ [json_value]({{site.url}}/docs/current/functions/json.html#json-value)
+ functions that were added to Trino this year.
+* The JSON output format was added to the `EXPLAIN` statement to provide
+ anonymised query plan to enable offline analysis.
+* It became possible to comment on tables, columns of tables, and even views for
+ various connectors. Support for setting comments on views was just introduced
+ recently and includes support for Hive and Iceberg. Table comments came to
+ Delta Lake, MySQL, BigQuery, Clickhouse and column comments came to Iceberg,
+ Accumulo, MongoDB, BigQuery, Clickhouse.
+* There were also a lot of new functions added like `to_base32`, `from_base32`,
+ `trim_array`, `trim`, and many others.
+
+### Performance improvements
+
+Despite all the hype about vectorization being a silver bullet to make databases
+go fast, the real speed comes from [better algorithms and better data structures
+that lead to lower resource consumption](https://www.youtube.com/clip/UgkxQwDYDS6evVJelNVjWAgrIhzg_Q-cAEyq).
+Here are a few that made their way into Trino this year:
+
+* Trino now offers improved performance for a variety of operations, including
+ complex join criteria pushdown to connectors, faster aggregations, faster
+ joins, and better performance for large clusters. We have also implemented
+ improvements specifically for aggregations with filters and for the Glue
+ metastore. In addition, we now support dynamic filtering for various connectors
+ and have faster query planning for Hive, Delta Lake, Iceberg, MySQL,
+ PostgreSQL, and SQL Server connectors.
+* Along with general performance optimizations, there have been a great deal of
+ query planning optimizations that lead to better performance for specific SQL
+ operators. These include faster `INSERT` queries, improved performance for
+ `LIKE` expressions and highly selective `LIMIT` queries, and enhanced
+ performance and reliability for `INSERT` and `MERGE` operations. We also made
+ performance improvements for `JOIN`, `UNION`, and `GROUP BY` queries, as well
+ as faster planning of queries with `IN` predicates. We have also added support
+ for correlated subqueries in `DELETE` queries.
+* There are also optimizations for specific SQL types’ performance like of
```

Comment:
There are also optimizations for specific SQL types’ performance such as string, decimal, MAP, and ROW types.